packages/blocks/src/api/utils.js

```diff
@@ -300,3 +257,29 @@ export function __experimentalGetBlockAttributesNamesByRole( name, role ) {
 			attributes[ attributeName ]?.__experimentalRole === role
 	);
 }
+
+/**
+ * Given attributes, remove any attributes with the given `role`.
+ *
+ * @param {string} name       The block's name.
+ * @param {Object} attributes The block's attributes.
+ * @param {string} role       The role of a block attribute.
+ *
+ * @return {Object} The attributes, with those matching the `role` removed.
+ */
+export function __experimentalRemoveAttributesByRole( name, attributes, role ) {
```

Comment:
This function feels a bit too specialised to me. It's better that complex functionality is achieved by composing together smaller functions rather than adding a new function. That way we're not shipping unnecessary bytes to the browser or having to maintain a large public API. 

Could callers use `_.omit` instead?

```js
const filteredAttributes = omit(
	block.attributes,
	getBlockAttributesNamesByRole( block.name, 'internal' )
);
```

https://devdocs.io/lodash~4/index#omit